.travis.yml

```diff
@@ -593,60 +593,6 @@ matrix:
-          before_script:
-            - nbbuild/travis/ant.sh $BUILD_OPTS clean
-            - nbbuild/travis/ant.sh $BUILD_OPTS build
-            - wget https://raw.githubusercontent.com/sormuras/bach/master/install-jdk.sh
-            - export TEST_JDK=`bash install-jdk.sh --feature 11 --license GPL --emit-java-home --silent | tail -1`
-            - export OPTS="-Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json $OPTS  -Dtest.nbjdk.home=$TEST_JDK -Dtest.run.args=--limit-modules=java.base,java.logging,java.xml,java.prefs,java.desktop,java.management,java.instrument,jdk.zipfs,java.scripting,java.naming -Dtest.bootclasspath.prepend.args=-Dno.netbeans.bootclasspath.prepend.needed=true"            
-          script:
-            #- ant $TEST_OPTS -f groovy/groovy test
-            #- ant $TEST_OPTS -f groovy/groovy.java test
-            - ant $TEST_OPTS -f groovy/groovy.editor test
-            - ant $TEST_OPTS -f groovy/groovy.gsp test
-            #- ant $TEST_OPTS -f groovy/groovy.kit test
-            - ant $TEST_OPTS -f groovy/groovy.support test
-            
-        - name: Test groovy modules on Java 17
-          jdk: openjdk8
-          env:
-            - BUILD_OPTS="-Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json -quiet -Dcluster.config=groovy -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false -Dtest-unit-sys-prop.ignore.random.failures=true"
-            - TEST_OPTS="-Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json -Dcluster.config=groovy -Djavac.compilerargs=-nowarn -Dbuild.compiler.deprecation=false -Dtest-unit-sys-prop.ignore.random.failures=true"
-          before_script:
-            - nbbuild/travis/ant.sh $BUILD_OPTS clean
-            - nbbuild/travis/ant.sh $BUILD_OPTS build
-            - wget https://raw.githubusercontent.com/sormuras/bach/master/install-jdk.sh
-            - export TEST_JDK=`bash install-jdk.sh --feature 17 --license GPL --emit-java-home --silent | tail -1`
-            - export OPTS="-Dmetabuild.jsonurl=https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/master/meta/netbeansrelease.json $OPTS  -Dtest.nbjdk.home=$TEST_JDK -Dtest.run.args=--limit-modules=java.base,java.logging,java.xml,java.prefs,java.desktop,java.management,java.instrument,jdk.zipfs,java.scripting,java.naming -Dtest.bootclasspath.prepend.args=-Dno.netbeans.bootclasspath.prepend.needed=true"            
-          script:
-            #- ant $TEST_OPTS -f groovy/groovy test
-            #- ant $TEST_OPTS -f groovy/groovy.java test
-            - ant $TEST_OPTS -f groovy/groovy.editor test
```

Comment:
can someone explain how the groovy tests actually work?

 - what does `OPTS` do? I don't think anything is picking this env var up
 - no JMS flags are set, which means this actually can't run on JDK 17 since it would fail with encapsulation exceptions
 - JAVA_HOME is in fact correctly set but i can see in the travis log lines like `[nb-javac] Support for javac --release has been added in Java9 ignoring it which further indicates that this is running on JDK 8

Reply:
Having been involved in creating the groovy matrix I will try to find some time to dig into this... stay tuned..

Reply:
@svenreimers awesome! If its of any help, thats my rough local test setup for experiments (this doesn't actually work due to some quotation issue, but you get the idea :)):

```bash
#!/bin/bash
set -e

export JAVA_HOME=/home/mbien/dev/java/jdk-11.0.15.9.1-corretto
export PATH=$JAVA_HOME/bin:$PATH

JMS_FLAGS=$(cat nbbuild/jms-config/base.flags nbbuild/jms-config/desktop.flags | xargs)

TEST_RUN_ARGS="-Dtest.run.args=\"--limit-modules=java.base,java.logging,java.xml,java.prefs,java.desktop,java.management,java.instrument,jdk.zipfs,java.scripting,java.naming $JMS_FLAGS\""

OPTS="-Dtest-unit-sys-prop.ignore.random.failures=true $TEST_RUN_ARGS"

echo $OPTS

ant $OPTS -f groovy/groovy.editor test
```